test/oas3/execute/style-serializer.js

```diff
@@ -1,9 +1,9 @@
-import { encodeDisallowedCharacters } from '../../../src/execute/oas3/style-serializer.js';
+import { valueEncoder } from '../../../src/execute/oas3/style-serializer.js';
 
 describe('OAS3 style serializer', () => {
```

Comment:
Why converting this test to testing `valueEncoder` instead of `encodeCharacters`? This test is actually testing only the `encodeCharacters` as it assumes that the input is always a string. If you want to test `valueEncoder` I'd create a separate test suite that tests it's aspects of coercion.

Reply:
This was done because `characterSet` was supposed to be only `unsafe` or `reserved` in `encodeCharacters` while the `escape` parameter should be handled by `valueEncoder`. Before in this test we were passing a `null` value as escape - the current `characterSet` would be the default `reserved` after the changes and the result would be encoded. We only return unencoded characters if `escape` is `false` in the `valueEncoder`. Before, this line took care of this case:
https://github.com/swagger-api/swagger-js/blob/f327d4bd28aa5d3ed646b412f8ec9422fe55441a/src/execute/oas3/style-serializer.js#L14-L16